src/index.js

```diff
@@ -19,13 +20,23 @@ afterAll(() => {
     }
 
     snapshotState.save();
+
+    if (commonSnapshotState) {
+      // Update common state so we get the report right with added/update/unmatched snapshots.
+      // Jest will display the "u" & "i" suggestion, plus displaying the right number of update/added/unmatched snapshots.
+      commonSnapshotState.unmatched += snapshotState.unmatched;
+      commonSnapshotState.matched += snapshotState.matched;
+      commonSnapshotState.updated += snapshotState.updated;
+      commonSnapshotState.added += snapshotState.added;
+    }
   });
 });
 
 function toMatchSpecificSnapshot(received, snapshotFile, testName) {
   const absoluteSnapshotFile = getAbsolutePathToSnapshot(this.testPath, snapshotFile);
 
-  const commonSnapshotState = this.snapshotState;
+  // store the common state to re-use it in "afterAll" hook.
+  commonSnapshotState = this.snapshotState;
```

Comment:
As I understand this will be replaced on every `toMatchSpecificSnapshot` call ?

Reply:
Indeed its is the only way we can get the original snapshotState. Not sure it worth it to add some checks to only store it the first time we get there. Wdty ?

Reply:
I have to debug it a bit to understand the whole picture =)